lib/saml_idp/xml_security.rb

```diff
@@ -97,10 +95,10 @@ def signature_algorithm(options)
         if options[:get_params] && options[:get_params][:SigAlg]
           algorithm(options[:get_params][:SigAlg])
         else
-          ref_elem = REXML::XPath.first(self, '//ds:Reference', { 'ds' => DSIG })
+          ref_elem = document.at_xpath('//ds:Reference | //Reference', DS_NS)
```

Comment:
does `//ds:Elem | //Elem` mean we are searching with and without a namespace?

Reply:
yes, exactly